lxd/storage/backend_lxd.go

```diff
@@ -729,12 +730,23 @@ func (b *lxdBackend) CreateInstanceFromBackup(srcBackup backup.Info, srcData io.
 		volumeConfig = srcBackup.Config.Volume.Config
 	}
 
+	// At this stage the DB record for the new volume isn't yet created.
+	// We have to set the UUID manually to allow passing it into the storage driver.
+	// It will get used later when creating the DB record.
+	volumeConfig["volatile.uuid"] = uuid.New().String()
```

Comment:
We should be trying to use a UUID from the backup config first no? So the volume maintains its original UUID?

Reply:
I was trying this first, but when you import the instance/volume multiple times like so `lxc import {c1.backup} c2`, then all the imports will end up with the same UUID. I thought it's not worth doing a lookup first in the DB, just assign a new UUID instead.

Reply:
Right makes sense, please can you add a comment here in the code explaining that specific scenario as the reason we do this. Ta

Reply:
Comment added.